portality/scripts/async_workflow_notifications.py

```diff
@@ -1,366 +1,11 @@
-    }
-
-    url = app.config.get("BASE_URL") + "/editor/your_applications"
-
-    es = models.Suggestion.query(q=assoc_age_query)
-    for bucket in es.get("aggregations", {}).get("assoc_ed", {}).get("buckets", []):    # loop through assoc_ed buckets
-        assoc_id = bucket.get("key")
-        idle = bucket.get("doc_count")
-
-        # Get the 'older than y weeks' count from nested aggregation
-        very_idle = bucket.get("older_weeks").get("buckets")[0].get('doc_count')         # only one bucket, so take first
-
-        # Pull the email address for our associate editor from their account
-        assoc = models.Account.pull(assoc_id)
-        try:
-            assoc_email = assoc.email
-        except AttributeError:
-            # There isn't an account for that id todo: should we tell someone about this?
-            continue
-
-        text = render_template('email/workflow_reminder_fragments/assoc_ed_age_frag', num_idle=idle, x_days=X_DAYS, num_very_idle=very_idle, y_weeks=Y_WEEKS, url=url)
-        _add_email_paragraph(assoc_email, assoc_id, text)
-
-
-# Helper functions
-def send_emails():
-    global emails_dict
-
-    for (email, (to_name, paragraphs)) in emails_dict.iteritems():
-        pre = 'Dear ' + to_name + ',\n\n'
-        post = '\n\nThe DOAJ Team\n\n***\nThis is an automated message. Please do not reply to this email.'
-        full_body = pre + '\n\n'.join(paragraphs) + post
-
-        app_email.send_mail(to=[email],
-                            fro=app.config.get('SYSTEM_EMAIL_FROM', 'feedback@doaj.org'),
-                            subject="DOAJ editorial reminders",
-                            msg_body=full_body)
-
-
-def _add_email_paragraph(addr, to_name, para_string):
-    """
-    Add a new email to the global dict which stores the email fragments, or extend an existing one.
-    :param addr: email address for recipient
-    :param to_name: name of recipient
-    :param para_string: paragraph to add to the email
-    """
-    global emails_dict
-
-    try:
-        (name, paras) = emails_dict[addr]
-        paras.append(para_string)
-    except KeyError:
-        emails_dict[addr] = (to_name, [para_string])
-
-
-# Main function for running all notification types in sequence
-def run_async_notifications():
-    """ Run through each notification type, then send emails """
-    # Create a request context to render templates
-    ctx = app.test_request_context()
-    ctx.push()
-
-    # Gather info and build the notifications
-    managing_editor_notifications()
-    editor_notifications()
-    associate_editor_notifications()
-
-    # Discard the context (the send mail function makes its own)
-    ctx.pop()
-
-    send_emails()
-
-# Run all if the script is called.
-if __name__ == '__main__':
-    run_async_notifications()
+from portality.tasks import async_workflow_notifications
+from portality.background import BackgroundApi
+
+if __name__ == "__main__":
+    user = app.config.get("SYSTEM_USERNAME")
```

Comment:
I'd do `app.config["SYSTEM_USERNAME"]` (fail if not declared) if the script can't really proceed without that config var being set.